### PR TITLE
Add some Ltac2.Constr.Unsafe combinators

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: user-contrib/Ltac2/Constr.v [hunk 1]
```diff
@@ -123,6 +132,142 @@ Module Case.
...
+      Array.iter (fun b => f n (Binder.type b)) tl;
+      let n := Array.fold_left g n tl in
+      Array.iter (f n) bl
+  | Array _u t def ty =>
+      f n ty;
+      Array.iter (f n) t;
+      f n def
+  end.
+
+Ltac2 binder_map (f : constr -> constr) (b : binder) : binder :=
```
[the remark this answers is not in the document]
Comment (SkySkimmer) [reply user-contrib/Ltac2/Constr.v:207]: It can get out of sync with the relevance so it's unsafe.

[the remark this answers is not in the document]
Comment (JasonGross) [reply user-contrib/Ltac2/Constr.v:207]: Should it be `Unsafe.Binder.map`?

[the remark this answers is not in the document]
Comment (SkySkimmer) [reply user-contrib/Ltac2/Constr.v]: I made the auxiliary functions local so we don't need to care where they go

**Comment by ppedrot** [comment]
@coqbot merge now

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**